package.json

```diff
@@ -73,6 +74,7 @@
     "number-allocator": "^1.0.7",
     "pump": "^3.0.0",
     "readable-stream": "^3.6.0",
+    "rfdc": "^1.3.0",
```

Comment:
always hesitant to add more dependencies. Could you explain the purpose of rfdc and collections, and why there isn't a suitable Node.js / JS builtin that can perform their tasks?

Reply:
rfdc is for clone that I mentioned at https://github.com/mqttjs/MQTT.js/pull/1301#discussion_r721431196
It is efficient and straightforward notation. It seems that the library is well maintained and the big number of weekly downloads. See Benchmarks https://www.npmjs.com/package/rfdc

AFAIK, Node.js/JS there is no straightforward way. 
One of the built-in way is as follows:

```js
function clone(a) {
   return JSON.parse(JSON.stringify(a));
}
```
https://stackoverflow.com/questions/6089058/nodejs-how-to-clone-an-object

Reply:
it seems to have created alot of issues with collections...

see there :
#https://github.com/Sunoo/homebridge-camera-ffmpeg/issues/1240
#https://github.com/NorthernMan54/homebridge-alexa/issues/481
#https://github.com/dgreif/ring/issues/844
and probably many more...